### PR TITLE
[OSD-18342] validate config on startup

### PR DESCRIPTION
**Pull request by clcollins**
* Adds validation functions for required configuration information
* Adds unit tests for validation functions
* Updates README.md to document configuration information

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -90,24 +101,164 @@ func init() {
...
+	}
+
+	err := errors.Join(configErrors...)
+	if err != nil {
+		valid = false
+		log.Print(err)
+	}
+
+	return valid
+}
```
Comment (mjlshen) [inline pkg/config/config.go:147]:
```suggestion
func (a *Config) Valid() bool {
	var configErrors error

	validationFunctions := []func(a *Config) []error{
		fieldsAreNotNil,
		hostFieldsAreParsable,
		passwordOrTokenExistIfUsernameProvided,
	}

	for _, f := range validationFunctions {
		configErrors := append(configErrors, f(a))
	}

	err := errors.Join(configErrors...)
	if configErrors != nil {
		log.Print(err)
	  return false
	}

	return true
}
```

What do you think about this?

**Review by mjlshen — CHANGES_REQUESTED** [review]
LGTM overall, small nit

File: pkg/config/config.go [hunk 2]
```diff
@@ -90,24 +101,159 @@ func init() {
...
+		configErrors = append(configErrors, f(a)...)
+	}
+
+	if configErrors != nil {
+		log.Print(errors.Join(configErrors...))
+		return false
+	}
+
+	return true
+}
```
Comment (clcollins) [inline pkg/config/config.go:142]: @mjlshen - I incorporated your changes, and adjusted this a little to simplify more.

**Review by mjlshen — APPROVED** [review]
/lgtm

**Pull request merged**